pkg/actuators/machine/actuator_test.go

```diff
@@ -3,184 +3,204 @@ package machine
+			invalidMachineScope: true,
+			awsError:            false,
 		},
 		{
-			name:    "Update machine event failed, reconciler's update failed",
-			machine: machine,
+			name: "Update machine event failed, reconciler's update failed",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Update(context.TODO(), machine)
 			},
-			event:    "Warning FailedCreate error",
-			awsError: true,
+			event:               "aws-actuator-testing-machine: reconciler failed to Update machine: AWS error",
+			invalidMachineScope: false,
+			awsError:            true,
 		},
 		{
-			name:    "Update machine event succeed",
-			machine: machine,
+			name: "Update machine event succeed and only one event is created",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Update(context.TODO(), machine)
+				actuator.Update(context.TODO(), machine)
 			},
-			event: "Normal Update Updated Machine aws-actuator-testing-machine",
+			event: "Updated Machine aws-actuator-testing-machine",
 		},
 		{
-			name:    "Delete machine event failed on invalid machine scope",
-			machine: machineInvalidProviderConfig,
+			name: "Delete machine event failed on invalid machine scope",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Delete(context.TODO(), machine)
 			},
-			event: "Warning FailedDelete aws-actuator-testing-machine: failed to create scope for machine: failed to get machine config: error unmarshalling providerSpec: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type v1beta1.AWSMachineProviderConfig",
+			event:               "aws-actuator-testing-machine: failed to create scope for machine: failed to create aws client: AWS client error",
+			invalidMachineScope: true,
+			awsError:            false,
 		},
 		{
-			name:    "Delete machine event failed, reconciler's delete failed",
-			machine: machine,
+			name: "Delete machine event failed, reconciler's delete failed",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Delete(context.TODO(), machine)
 			},
-			event:    "Warning FailedDelete error",
-			awsError: true,
+			event:               "aws-actuator-testing-machine: reconciler failed to Delete machine: AWS error",
+			invalidMachineScope: false,
+			awsError:            true,
 		},
 		{
-			name:    "Delete machine event succeed",
-			machine: machine,
+			name: "Delete machine event succeed",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Delete(context.TODO(), machine)
 			},
-			event: "Normal Delete Deleted machine aws-actuator-testing-machine",
+			event:               "Deleted machine aws-actuator-testing-machine",
+			invalidMachineScope: false,
+			awsError:            false,
 		},
 	}
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.TODO()
```

Comment:
You'll want a new gomega for each subtest so you can fail the subtests separately from the main test
```suggestion
			ctx := context.TODO()
			gs := NewWithT(t)
```